src/applications/apps.js

```diff
@@ -350,12 +352,78 @@ function sanitizeArguments(
     registration.customProps = customProps;
   }
 
-  if (typeof registration.loadApp !== "function") {
-    const { loadApp } = registration;
-    registration.loadApp = () => Promise.resolve(loadApp);
+  registration.loadApp = sanitizeLoadApp(registration.loadApp);
+  registration.customProps = sanitizeCustomProps(registration.customProps);
+  registration.activeWhen = sanitizeActiveWhen(registration.activeWhen);
+
+  return registration;
+}
+
+function sanitizeLoadApp(loadApp) {
+  if (typeof loadApp !== "function") {
+    return () => Promise.resolve(loadApp);
   }
 
-  if (!registration.customProps) registration.customProps = {};
+  return loadApp;
+}
 
-  return registration;
+function sanitizeCustomProps(customProps) {
+  return customProps ? customProps : {};
+}
+
+function sanitizeActiveWhen(activeWhen) {
+  let activeWhenArray = Array.isArray(activeWhen) ? activeWhen : [activeWhen];
+
+  return (location) =>
+    activeWhenArray.some((activeWhenOrPath) => {
+      const activeWhen =
+        typeof activeWhenOrPath === "function"
+          ? activeWhenOrPath
+          : pathToActiveWhen(activeWhenOrPath);
```

Comment:
It would be better for performance to call `pathToActiveWhen` once upfront instead of every time the activity function runs:

```js
let activeWhenArray = Array.isArray(activeWhen) ? activeWhen : [activeWhen]; 
activeWhenArray = activeWhenArray.map(activeWhenOrPath => typeof activeWhenOrPath === "function" ? activeWhenOrPath : pathToActiveWhen(activeWhenOrPath))

return location => activeWhenArray.some(activeWhen => activeWhen(location));
```

This way we don't have to run `toDynamicPathValidatorRegex` every time the route changes.